Readme.md

```diff
@@ -1,5 +1,154 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("path/to/example.bnf");
+```
+
+To parse the input you should use `flux_bnf::bnf::parse`
+```rust
+let mut lexer = flux_bnf::bnf::parse(bnf_input).unwrap();     
+```
+
+## How to set Lexer Rules
+Now that you have `Lexer` we can use that to now set some rules
+
+First you will need to add some rules to the BNF before we can set the rules
```

Comment:
The wording here is kind of confusing - you refer to BNF rules, as well as lexer token culling rules. It's also worth noting that at this point in the readme, the reader has not seen any BNF. You might want to consider changing the order or saying something like "You'll need to write a BNF file describing the syntax, detailed below" with a link to the BNF subsection.